central/sensor/service/connection/messagestream/sizing_events.go

```diff
@@ -0,0 +1,29 @@
+package messagestream
+
+import (
+	"math"
+
+	"github.com/stackrox/rox/central/metrics"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/reflectutils"
+)
+
+type sizingEventStream struct {
+	stream  CentralMessageStream
+	maxSeen map[string]float64
+}
+
+func (s *sizingEventStream) Send(msg *central.MsgToSensor) error {
+	typ := reflectutils.Type(msg.GetMsg())
+	gaugeValue := math.Max(s.maxSeen[typ], float64(msg.Size()))
+	metrics.ObserveSentSize(typ, float64(msg.Size()))
+	metrics.SetGRPCLastMessageSizeGauge(typ, float64(msg.Size()))
+	metrics.SetGRPCMaxMessageSizeGauge(typ, gaugeValue)
+	s.maxSeen[typ] = gaugeValue
+	return s.stream.Send(msg)
+}
```

Comment:
Just to confirm, we no longer need to have a mutex and lock the `maxSeen` map, do we?

Reply:
Did we ever? These wrappers are synchronous and will process one gRPC message at a time.

Reply:
I think it is, it runs in single goroutne `runSend` which reads from a messages channel: https://github.com/stackrox/stackrox/blob/master/central/sensor/service/connection/connection_impl.go#L200-L215